server/etcdserver/api/snap/snapshotter.go

```diff
@@ -88,7 +89,9 @@ func (s *Snapshotter) save(snapshot *raftpb.Snapshot) error {
 	spath := filepath.Join(s.dir, fname)
 
 	fsyncStart := time.Now()
+	cancel := watchdog.Register("save v2 snapshot")
```

Comment:
I think we need to decide the naming better. 

a) ```
  endActivity := wd.StartActivity("saving v2 snapshot")
  endActivity()
```

b)  ```
  markAsDone := wd.Start("saving v2 snapshot")
  markAsDone()
```

c) ```
  markAsDone := wd.Notify("saving v2 snapshot")
  markAsDone()
```

d) ```
 endScope := wd.StartScope("saving v2 snapshot")
 endScope()
```

------------------------------------------------------------

We might also have a synthactic sugar:

```
or watchdog.executeInScope("saving v2 snapshot", func() { ...} );
```

I don't like 'cancel' as it suggests an interruption of an existing process.. and it's not. 

-----------
I hoped we will settle this (API / naming) in a document rather than keep updating the PR, that is relatively expensive.

Reply:
I like the suggestion of `synthactic sugar`. I imagine it's similar to bboltDB's `View` and `Update` method

Reply:
`StartXXX` seems not good to me, it doesn't start any activity, instead essentially it just registers the activity to watchdog. So how about Register and unregister? 

```
unregister := wd.Register("saving v2 snapshot")
unregister()
```

Users can also call the syntactic sugar as you proposed below,
```
wd.Execute("saving v2 snapshot", fn)
```

Please also see the updated doc https://docs.google.com/document/d/1U9hAcZQp3Y36q_JFiw2VBJXVAo2dK2a-8Rsbqv3GgDo/edit#heading=h.3oeryohw1c9o